linkml_runtime/utils/schemaview.py

```diff
@@ -51,11 +53,14 @@ def _closure(f, x, reflexive=True, depth_first=True, **kwargs):
             i = todo[0]
             todo = todo[1:]
         visited.append(i)
-        vals = f(i)
-        for v in vals:
-            if v not in visited:
-                todo.append(v)
-                if v not in rv:
+        if kwargs.get('enum', False):
+            vals = f(i, kwargs.get('enum'))
+        else:
+            vals = f(i)
```

Comment:
If this is necessary I would suggest just unconditionally passing all all keyword arguments to `f` instead of adding enum-specific stuff to this generic function:

```python
vals = f(i, **kwargs)
```

**But** I don't think that's really necessary anyway. See my later comment on `permissible_value_ancestors`.

Reply:
fixed.